crates/accelerate/src/target_transpiler/errors.rs

```diff
@@ -31,4 +31,23 @@ impl Display for TargetKeyError {
     }
 }
 
-impl Error for TargetKeyError {}
+/// Error thrown when an invalid instruction is added to the [Target]
+#[derive(Debug)]
+pub struct TargetInvalidInstError {
```

Comment:
We already depend on this error: https://docs.rs/thiserror/latest/thiserror/ you can use it to clean up most of the boilerplate code here.